### PR TITLE
Validator and mutator webhooks

### PR DESCRIPTION
**Pull request by lsroe**
**What this PR does / why we need it**:

PR provides mutator and validator webhooks for the falco extension definitions in the shoot specs

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: pkg/admission/mutator/shoot_mutator.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+	setFalcoCtl(falcoConf)
+
+	setGardenerRules(falcoConf)
+
+	setCustomWebhook(falcoConf)
+
+	return s.updateFalcoConfig(new, falcoConf)
+
+	// oldNamedResources := map[string]int{}
```
Comment (marwinski) [inline pkg/admission/mutator/shoot_mutator.go:167]: can you delete this section

File: pkg/admission/mutator/shoot_mutator.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+	extensionType := "shoot-falco-service"
+	for i, ext := range shoot.Spec.Extensions {
+		if ext.Type == extensionType {
+			return &shoot.Spec.Extensions[i]
+		}
+	}
+	return nil
+}
+
+func (s *shoot) extractFalcoConfig(shoot *gardencorev1beta1.Shoot) (*service.FalcoServiceConfig, error) {
```
Comment (marwinski) [inline pkg/admission/mutator/shoot_mutator.go:223]: can we move this to a utils class? Similar coding is used in other places as well.

File: pkg/admission/validator/shoot_validator.go [hunk 3]
```diff
@@ -30,66 +35,152 @@ type shoot struct {
...
-	return nil, nil
-}
-
-// findExtension returns shoot-dns-service extension.
+// findExtension returns shoot-falco-service extension.
 func (s *shoot) findExtension(shoot *core.Shoot) *core.Extension {
+	extensionType := "shoot-falco-service"
 	for i, ext := range shoot.Spec.Extensions {
-		if ext.Type == service.ExtensionType {
+		if ext.Type == extensionType {
```
Comment (marwinski) [inline pkg/admission/validator/shoot_validator.go:169]: Constants.ExtensionType

File: pkg/admission/validator/shoot_validator.go [hunk 4]
```diff
@@ -30,66 +35,152 @@ type shoot struct {
...
 			return &shoot.Spec.Extensions[i]
 		}
 	}
 	return nil
 }
-*/
+
+func (s *shoot) extractFalcoConfig(shoot *core.Shoot) (*service.FalcoServiceConfig, error) {
```
Comment (marwinski) [inline pkg/admission/validator/shoot_validator.go:176]: Duplicate, see above

**Review by marwinski — APPROVED** [review]
some minor comments, I think it would be ok to go ahead and fix in due course.

File: pkg/admission/mutator/shoot_mutator.go (hunk 1 above)
Comment (lsroe) [reply pkg/admission/mutator/shoot_mutator.go:167]: Removed

File: pkg/admission/mutator/shoot_mutator.go (hunk 2 above)
Comment (lsroe) [reply pkg/admission/mutator/shoot_mutator.go:223]: Needs to stay as shoot is different struct for validator and mutator

File: pkg/admission/validator/shoot_validator.go (hunk 4 above)
Comment (lsroe) [reply pkg/admission/validator/shoot_validator.go:176]: Needs to stay as shoot is different struct for validator and mutator

**Pull request merged**